pylint/lint/pylinter.py

```diff
@@ -379,13 +380,13 @@ def load_plugin_configuration(self) -> None:
         hook, if exposed, and calls it to allow plugins to configure specific
         settings.
         """
-        for modname in self._dynamic_plugins:
-            try:
-                module = astroid.modutils.load_module_from_name(modname)
-                if hasattr(module, "load_configuration"):
-                    module.load_configuration(self)
-            except ModuleNotFoundError as e:
-                self.add_message("bad-plugin-value", args=(modname, e), line=0)
+        for modname, module_or_error in self._dynamic_plugins.items():
```

Comment:
Just for my own understanding and to recap.

The issue is fixed because rather than trying to import twice, one of which was after `sys.path` was updated, we now try to load a plugin once and work with that result in this function. Right?

Reply:
Yes, that's right; we try to load before the config file `init-hook` is run and just remember the result. I'll add an inline comment to give detail here also.